rusk/src/lib/http/chain.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+pub mod graphql;
+
+use node::database::rocksdb::Backend;
+use node::network::Kadcast;
+
+use juniper::EmptyMutation;
+use juniper::EmptySubscription;
+use juniper::Variables;
+use std::sync::Arc;
+
+use graphql::{DbContext, Query};
+
+use crate::chain::RuskNode;
+use crate::http::{DataType, WsRequest, WsResponse};
+
+use super::event::WsTarget;
+
+type Schema = juniper::RootNode<
+    'static,
+    Query,
+    EmptyMutation<DbContext>,
+    EmptySubscription<DbContext>,
+>;
+
+impl RuskNode {
+    pub(crate) async fn handle_request(
+        &self,
+        request: WsRequest,
+    ) -> WsResponse {
```

Comment:
I'm not sure this belongs here. Either `RuskNode` should be here, or this in `lib/chain.rs`. Feels disconnected like this.

Reply:
Implement it here will allow us to not include the whole `http` stuff easily